Fix cryptic file validation errors

ahukkanen opened the pull request:
#### :tophat: What? Why?
Currently the file validation messages are rather cryptic for the general user because they are showing the content type matching pattern regular expressions converted into strings.

To make it understandable for the general public, we want to show the file extensions instead of the content types.

#### :pushpin: Related Issues
- Fixes #9185

#### Testing
- Enable attachments for proposals
- Try to upload an attachment which is not allowed, e.g. a `.zip` file is not allowed by default
- See a better error message

andreslucena commented:
This is a great improvement on what we have for sure!

1. I think we can tune it a bit more. Now it says "file should be one of *.bmp, *.gif, *.jpeg, *.jpg, *.pdf, *.png, *.rtf, *.txt". Even though the second part is much better, it still sounds a bit too technical. The first part ("should be one of") sounds like Set Theory, and the second part (*.bmp) has regular expressions. Could you change it to something like this 🙏🏽?

> Only files with the following extensions are allowed: gif jpeg jpg png bmp

2. In the avatar file upload in the participant account, there's something that breaks this: 

![image](https://user-images.githubusercontent.com/717367/190430930-4c2bbfcc-1501-4cb5-ba70-56e98cc4b252.png)

As far as I could find is the only one with mixed formats. 

3. In the file upload attachment of a participatory process, the file extensions of the error doesn't correspond with the ones from the help text. 

![image](https://user-images.githubusercontent.com/717367/190431554-47040069-4428-4dfb-858d-0120b03045e5.png)

4. On the private participants census upload CSV it says `text/csv`
 
![image](https://user-images.githubusercontent.com/717367/190431779-b7a521d2-28e2-4081-9586-a93b3825ac8a.png)

(Mind that the help text doesn't correspond and talks about images, but that's out of the scope of the current PR)

ahukkanen commented:
@andreslucena All issues fixed that I could fix in this PR. The rest will be fixed by another PR #9623. See explanations below.

> 1. I think we can tune it a bit more. Now it says "file should be one of *.bmp, *.gif, *.jpeg, *.jpg, *.pdf, *.png, *.rtf, _.txt". Even though the second part is much better, it still sounds a bit too technical. The first part ("should be one of") sounds like Set Theory, and the second part (_.bmp) has regular expressions. Could you change it to something like this 🙏🏽?
> > Only files with the following extensions are allowed: gif jpeg jpg png bmp

Done. Also changed the other similar error messages to match this format.

> 2. In the avatar file upload in the participant account, there's something that breaks this:
> 
> As far as I could find is the only one with mixed formats.

The mixed format is because Decidim produces unknown MIME types currently. E.g. `image/bmp` is not known for MiniMime (instead, it knows `image/x-bmp`). And also `image/jpg` is not a valida MIME type anywhere.

This has been a persistent issue in Decidim for years but I am fixing it in another PR #9623.

In particular it is this line that will fix this issue:
https://github.com/decidim/decidim/blob/4b9d5399b72cb6594484866bbfcce7985a57f651/decidim-core/app/uploaders/decidim/image_uploader.rb#L11

Do note that this same issue will still persist in case the system administrator has configured MIME types unknown for MiniMime. But I would see that as misconfiguration rather than a bug, and it'd be good to show those "extra" unknown MIME types for the user in case that happens to make it easier to identify potential issues regarding those.

> 3. In the file upload attachment of a participatory process, the file extensions of the error doesn't correspond with the ones from the help text.

This was because the file validation wasn't context aware whether the validation is happening at the admin panel or the participant side of the system which can have different allowed extensions through the system settings.

I have fixed this issue in the latest revisions.

> 4. On the private participants census upload CSV it says `text/csv`

This was because initially I only applied these changes to the validations to the upload validations happening for files that are attached to the records through uploaders. This was not applied in case you just used `validates :some_temp_file, file_content_type: { allow: ["text/csv"] }` which happens e.g. for the described use case (votings census).

I had to do [a bit of a "hack"](https://github.com/decidim/decidim/pull/9663/commits/c3bcaf5fbff381af42119fd49a5efc4fcfe02608) to also fix the messages for such direct validations by overriding one method from `ActiveModel::Validations::FileContentTypeValidator`. But anyways, now it should work for both cases.

andreslucena commented:
>> The mixed format is because Decidim produces unknown MIME types currently. E.g. image/bmp is not known for MiniMime (instead, it knows image/x-bmp). And also image/jpg is not a valida MIME type anywhere.
>
> This has been a persistent issue in Decidim for years but I am fixing it in another PR https://github.com/decidim/decidim/pull/9623.

Great! I'll check that one after merging this one. 

> Do note that this same issue will still persist in case the system administrator has configured MIME types unknown for MiniMime. But I would see that as misconfiguration rather than a bug, and it'd be good to show those "extra" unknown MIME types for the user in case that happens to make it easier to identify potential issues regarding those.

Ok, I agree.

andreslucena reviewed (approved):
I've rechecked all my comments and are all fixed 👍🏽 👍🏽 
Code-wise everything is great, documented, and tested. 💯
